python/lance/tests/types/pandas/test_image.py

```diff
@@ -0,0 +1,243 @@
+
+
+class TestDtype(base.BaseDtypeTests):
+
+    @pytest.mark.skip(reason="not a problem")
+    def test_is_not_string_type(self, dtype):
+        # super class didn't assert the value
+        assert not is_string_dtype(dtype)
+
+    def test_is_not_object_type(self, dtype):
+        # super class didn't assert the value
+        assert not is_object_dtype(dtype)
+
+
+class TestInterface(base.BaseInterfaceTests):
+    def test_array_interface(self, data):
+        result = np.array(data)
+        assert result[0] == data[0].uri
+
+        result = np.array(data, dtype=object)
+        expected = np.array([img.uri for img in list(data)], dtype=object)
+        tm.assert_numpy_array_equal(result, expected)
+
+
+class TestConstructors(base.BaseConstructorsTests):
+    pass
+
+
+class TestGetitem(base.BaseGetitemTests):
+    pass
+
+
+class TestSetitem(base.BaseSetitemTests):
+    pass
+
+
+class TestIndex(base.BaseIndexTests):
+    pass
+
+
+class TestMissing(base.BaseMissingTests):
+    pass
+
+
+class TestNoReduce(base.BaseNoReduceTests):
+    @pytest.mark.parametrize("skipna", [True, False])
+    def test_reduce_series_numeric(self, data, all_numeric_reductions, skipna):
+        op_name = all_numeric_reductions
+
+        if op_name in ["min", "max"]:
+            return
+
+        ser = pd.Series(data)
+        with pytest.raises(TypeError):
+            getattr(ser, op_name)(skipna=skipna)
+
+
+class TestCasting(base.BaseCastingTests):
+    def test_astype_str(self, data):
+        result = pd.Series(data[:5]).astype(str)
+        expected = pd.Series([x.uri for x in data[:5]], dtype=str)
+        self.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "nullable_string_dtype",
+        [
+            "string[python]",
+            pytest.param(
+                "string[pyarrow]", marks=td.skip_if_no("pyarrow", min_version="1.0.0")
+            ),
+        ],
+    )
+    def test_astype_string(self, data, nullable_string_dtype):
+        # GH-33465
+        result = pd.Series(data[:5]).astype(nullable_string_dtype)
+        expected = pd.Series([x.uri for x in data[:5]], dtype=nullable_string_dtype)
+        self.assert_series_equal(result, expected)
+
+
+class TestComparisonOps(base.BaseComparisonOpsTests):
+    def _compare_other(self, ser, data, op, other):
+        op_name = f"__{op.__name__}__"
+        result = getattr(ser, op_name)(other)
+        expected = getattr(ser.astype(object), op_name)(other).astype("boolean")
+        self.assert_series_equal(result, expected)
+
+    def test_compare_scalar(self, data, comparison_op):
+        ser = pd.Series(data)
+        self._compare_other(ser, data, comparison_op, "abc")
+
+    def test_compare_array(self, data, comparison_op):
+        ser = pd.Series(data)
+        other = pd.Series([str(data[0])] * len(data))
+        self._compare_other(ser, data, comparison_op, other)
+
+
+class TestParsing(base.BaseParsingTests):
```

Comment:
What are those empty class used for?

Reply:
Basically these are test suites for every pandas ExtensionArray. When you make a new one, you just override the fixtures and override/skip tests that are specific to your ExtensionArray